### PR TITLE
Simplify view_generic and back_to_main

### PR DESCRIPTION
**Pull request by orangejenny**
https://github.com/dimagi/commcare-hq/pull/24946#pullrequestreview-267383321

Followup for https://github.com/dimagi/commcare-hq/pull/24967

### CODE REVIEW

File: corehq/apps/app_manager/views/utils.py [hunk 1]
```diff
@@ -63,56 +63,40 @@ def back_to_main(request, domain, app_id=None, module_id=None, form_id=None,
...
-        view_name = page
-
-    return HttpResponseRedirect(
-        "%s%s" % (
-            reverse(view_name, args=args),
-            "?%s" % urlencode(params) if params else ""
-        )
-    )
+    if form is not None:
+        view_name = 'view_form' if form.no_vellum else 'form_source'
```
Comment (mkangia) [inline corehq/apps/app_manager/views/utils.py]: woah! how did that get into the code base, that was an incorrect indent right?

File: corehq/apps/app_manager/views/view_generic.py [hunk 2]
```diff
@@ -59,73 +59,67 @@ def view_generic(request, domain, app_id, module_id=None, form_id=None,
...
+        try:
+            form = app.get_form(form_unique_id)
+        except FormNotFoundException:
+            raise Http404()
+        form_id = form.id
 
-    except (ModuleNotFoundException, FormNotFoundException):
-        return bail(request, domain, app_id)
+    if form is not None and module is None:
+        # this is the case where only the form_unique_id is given
```
Comment (mkangia) [inline corehq/apps/app_manager/views/view_generic.py]: thank you for adding this, whoever did :)

**Review by mkangia — APPROVED** [review]

**Review by emord — APPROVED** [review]
nice

File: corehq/apps/app_manager/views/view_generic.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/app_manager/views/view_generic.py]: @biyeun !

File: corehq/apps/app_manager/views/utils.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/app_manager/views/utils.py]: Hmm? It's not an incorrect indent, I'm just removing the `if app` checks and unindenting. But this diff might have come out weird.

**Pull request merged**